File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetscommercecustomer.go
package shared

import (
	"time"
)

type CodatDataContractsDatasetsCommerceCustomer struct {
	Addresses          []CodatDataContractsDatasetsCommerceAddress `json:"addresses,omitempty"`
	CreatedDate        *time.Time                                  `json:"createdDate,omitempty"`
	CustomerName       *string                                     `json:"customerName,omitempty"`
	DefaultCurrency    *string                                     `json:"defaultCurrency,omitempty"`
	EmailAddress       *string                                     `json:"emailAddress,omitempty"`
	ID                 *string                                     `json:"id,omitempty"`
	ModifiedDate       *time.Time                                  `json:"modifiedDate,omitempty"`
	Note               *string                                     `json:"note,omitempty"`
	Phone              *string                                     `json:"phone,omitempty"`
	SourceModifiedDate *time.Time                                  `json:"sourceModifiedDate,omitempty"`
}

File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_commerce_customer.py
import dataclasses
import dateutil.parser
from ..shared import codat_datacontracts_datasets_commerce_address as shared_codat_datacontracts_datasets_commerce_address
from dataclasses_json import dataclass_json
from datetime import datetime
from marshmallow import fields
from sdk import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class CodatDataContractsDatasetsCommerceCustomer:
    addresses: Optional[list[shared_codat_datacontracts_datasets_commerce_address.CodatDataContractsDatasetsCommerceAddress]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('addresses') }})
    created_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('createdDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    customer_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('customerName') }})
    default_currency: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('defaultCurrency') }})
    email_address: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('emailAddress') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    modified_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('modifiedDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    note: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('note') }})
    phone: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('phone') }})
    source_modified_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sourceModifiedDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    